### PR TITLE
Fix for API services bug #6894

### PR DESCRIPTION
**Pull request by Dubby85**
Fixed an issue with the output including all services from a host that had a single matched service  regardless of whether they matched the filter.

Bug report: https://github.com/librenms/librenms/issues/6894

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

fixes: #6894

### CODE REVIEW

**Comment by murrant** [comment]
It might be nicer to build the query like this (partial untested example)
```php
$where = array();
$params = array();

if (isset($_GET['state'])) {
    $where[] ='`service_status`=?';
    $params[] = $_GET['state'];
    $where[] = "`service_disabled`='0'";
    $where[] = "`service_ignore`='0'";
}

$query = 'SELECT * FROM `services`';
if(!empty($where)) {
    $query .= ' WHERE ' . implode(' AND ', $where);
}
$query .= ' GROUP BY S.service_ip ORDER BY S.service_ip';

$services = dbFetchRows($query, $params);
```

Try to use ? for any supplied variables.

**Comment by Dubby85** [comment]
Thanks @murrant it worked out of the box, I have committed an updated version based on your suggestion.

File: html/includes/api_functions.inc.php [hunk 1]
```diff
@@ -1510,53 +1511,50 @@ function list_services()
...
-            $devicewhere  = " AND `service_type` LIKE ?";
-            $sql_param[1] = $_GET['type'];
-        }
-
-        $services[] = dbFetchRows("SELECT * FROM `services` WHERE `device_id` = ?".$devicewhere, $sql_param);
+    if (!empty($where)) {
+        $query .= ' WHERE ' . implode(' AND ', $where);
     }
+    $query .= ' ORDER BY service_ip';
+    $services[] = dbFetchRows($query, $params);
```
Comment (murrant) [inline html/includes/api_functions.inc.php]: Odd that $services is a double nested array.

**Comment by murrant** [comment]
Thanks for the fix, much better now.

**Pull request merged**